rules/integrations/aws/privilege_escalation_attached_malicious_lambda_layer.toml

```diff
@@ -0,0 +1,57 @@
+[metadata]
+creation_date = "2021/10/03"
+maturity = "production"
+updated_date = "2021/10/03"
+integration = "aws"
+
+[rule]
+author = ["Austin Songer"]
+description = """
+Identifies when a user attaches a Lambda layer to an existing function to override a library that is in use by the
+function. Attackers can use the Lambda layer to escalate privileges executing API calls using the functions IAM role. 
+This would give an adversary access to the privileges associated with the Lambda service role attached to that function.
+"""
+false_positives = [
+    """
+    Lambda Layer being attached may be performed by a system administrator. Verify whether the user identity, user 
+    agent, and/or hostname should be making changes in your environment. Lambda Layer being attached from unfamiliar
+    users should be investigated. If known behavior is causing false positives, it can be exempted from the rule.
+    """,
+]
+from = "now-25m"
+index = ["filebeat-*", "logs-aws*"]
+language = "kuery"
+license = "Elastic License v2"
+name = "AWS Lambda Layer Attached"
+note = """## Config
+
+The AWS Fleet integration, Filebeat module, or similarly structured data is required to be compatible with this rule."""
+references = [
+    "https://docs.aws.amazon.com/lambda/latest/dg/API_UpdateFunctionConfiguration.html",
+    "https://rhinosecuritylabs.com/aws/aws-privilege-escalation-methods-mitigation-part-2/",
+]
+risk_score = 21
+rule_id = "59ada119-64f4-466f-be04-7a123f5a107a"
+severity = "low"
+tags = ["Elastic", "Cloud", "AWS", "Continuous Monitoring", "SecOps", "Identity and Access"]
+timestamp_override = "event.ingested"
+type = "query"
+
+query = '''
+event.dataset:aws.cloudtrail and event.action:(UpdateFunctionConfiguration and 
+AttachUserPolicy) and event.outcome:success
```

Comment:
I don't think this will match the event, as it would require the event to have both values on it

Reply:
If we need to match two events, I recommend you to use the eql search sequence, if you need any help on this, let me know
Ref: https://www.elastic.co/guide/en/elasticsearch/reference/master/eql.html#eql-search-sequence